doc/Architecture.md

```diff
@@ -280,5 +280,115 @@ The offchain code that constructs the DeNS database will always treat the _most
+  [oura](https://github.com/txpipe/oura/#readme) which filters events on the
+  blockchain according to patterns and outputs the values to a "sink" which
+  updates our own database.
+  Note that we could also use
+  [marconi](https://github.com/input-output-hk/marconi), but it appears to be
+  under heavy development.
+
+- [Interacting with ogmios](https://ogmios.dev/) which provides a convenient
+  websockets to interact with the blockchain node, and we could update our
+  database based on the information obtained by the websockets.
+
+As an ASCII diagram, we envision this to be
+
+```
+                  
+                   Listens for events
+ --------------                         ---------------          --------------
+| Cardano node | --------------------- | Chain indexer | ------ | SQL database |
+ --------------                         ---------------          --------------
+            \                                 | 
+             \                                |
+               \                              | Query UTxOs
+      Submit txs \                            |
+                  --\                         | 
+                     \                   ------------ 
+                      \---------------- | Tx builder |        
+                                         ------------               
+```
+
+where the lines denote communication, and the square boxes denote programs.
+
+
+### Providing DNS records
+We will piggyback off of existing DNS systems.
+In particular, we will adapt the existing full-featured DNS system [BIND 9](https://www.isc.org/bind/) for our purposes.
+Note that BIND 9 is the first, oldest, and most commonly deployed DNS solution, so this has the immediate benefit that network engineers will already be familiar with its deployment to ease adoption of DeNS.
+
+
+BIND 9 is configured by a file often known as `named.conf`.
+Adding the initial set of DNS records to BIND 9 is straightforward -- one
+updates the `named.conf` to include a zone file (set of DNS records) with the
+associated DNS name. See the documentation for
+[details](https://bind9.readthedocs.io/en/latest/chapter3.html#primary-authoritative-name-server).
+Updates to the database can be done dynamically with [nsupdate](https://bind9.readthedocs.io/en/latest/manpages.html#std-iscman-nsupdate)
+    or by directly following the [dynamic updates in DNS protocol](https://datatracker.ietf.org/doc/html/rfc2136.html).
+
+One wrinkle with using a traditional DNS system for DeNS is that the blockchain
+    is the single authoritative source of ownership for all records as the onchain
+    smart contracts guarantee that every domain has a unique name or label.
+Moreover, all DNS records should be immediately wholly accessible (TODO: don't
+    know how to word this?) via the blockchain (indirectly by Arweave); so the
+    DNS server should never need to recursively resolve queries by contacting
```

Comment:
Well... it probably does need to contact other name servers (for mirrored traditional DNS records) to verify that the records haven't changed after the TTL has lapsed...

But let's just ignore that for right now.